### PR TITLE
Feat(agnoster) Add terraform workspace name prompt support. Plugins terraform-workspace-prompt)

### PR DESCRIPTION
**Pull request by ocontant**
## Standards checklist:

- [X] The PR title is descriptive.
- [X] The PR doesn't replicate another PR which is already open.
- [X] I have read the contribution guide and followed all the instructions.
- [X] The code follows the code style guide detailed in the wiki.
- [X] The code is mine or it's from somewhere with an MIT-compatible license.
- [X] The code is efficient, to the best of my ability, and does not waste computer resources.
- [X] The code is stable and I have tested it myself, to the best of my abilities.
- [X] If the code introduces new aliases, I provide a valid use case for all plugin users down below.

## Changes:

- Add support to display terraform workspace name in Agnoster theme prompt.
- The change is designed to support the plugin terraform-workspace-prompt

## Other comments:
There is currently no plugin that support to easily and aesthetically display terraform workspace in zsh prompt, using agnoster theme prompt engine. 

I added a terraform prompt logic, to uniformely support terraform workspace name to be displayed in the agnoster prompt.

I'm currently using this code via /.oh-my-zsh/custom/... structure.

Dependency: 
- [terraform-workspace-prompt](https://github.com/ocontant/terraform-workspace)
  - ohmyzsh PR: Plugins: terraform-workspace-prompt
  
  

If you find this contribution useful, feel free to merge. 

Best Regards,

### CODE REVIEW

**Review by soheilwikey1 — APPROVED** [review]
Review

File: plugins/terraform-workspace-prompt/README.md
Comment (carlosala) [inline plugins/terraform-workspace-prompt/README.md:1]: Could we simplify this file to only have `omz` stuff. It should be similar to https://github.com/ohmyzsh/ohmyzsh/tree/master/plugins/virtualenv readme file.

File: plugins/terraform-workspace-prompt/_terraform
Comment (carlosala) [inline plugins/terraform-workspace-prompt/_terraform:1]: This file should be deleted, it's already present in `terraform` plugin.

File: plugins/terraform-workspace-prompt/terraform-workspace.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    __TERRAFORM_WORKSPACE_CACHE=""
+  fi
+}
+
+# Hooks to call the update function when the working directory changes or before each prompt is displayed
+autoload -Uz add-zsh-hook
+add-zsh-hook chpwd __update_terraform_workspace_prompt
+add-zsh-hook precmd __update_terraform_workspace_prompt
+
+complete -o nospace -C $(which terraform) terraform
```
Comment (carlosala) [inline plugins/terraform-workspace-prompt/terraform-workspace.plugin.zsh:35]: This is not needed, as there's already zsh completion present in `terraform` file.

**Review by carlosala — CHANGES_REQUESTED** [review]

**Comment by carlosala** [comment]
Hi! This PR doesn't seem to get a lot of traction, and it seems stale. Closing, thanks for the contribution!

**Pull request closed without merging**